feat: troubleshoot write errors (#3003). Fix link in Flux reference. (#3104)
influxdata/docs-v2: content/influxdb/cloud/write-data/troubleshoot.md

  If some of your data did not write to the bucket, see how to [troubleshoot rejected points](#troubleshoot-rejected-points).
    {{% /note %}}

- `400` **Bad request**: The line protocol data in the request was malformed.
   The response body contains the first malformed line in the data. All request data was rejected and not written.
- `401` **Unauthorized**: May indicate one of the following:
  - `Authorization: Token` header is missing or malformed.
  - API token value is missing from the header.
  - API token does not have sufficient permissions to write to the organization and bucket.
- `404` **Not found**: A requested resource (e.g. an organization or bucket) was not found. The response body contains the requested resource type, e.g. "organization", and resource name.
- `413` **Request entity too large**: The payload exceeded the 50MB limit. All request data was rejected and not written.
- `429` **Too many requests**: API token is temporarily over quota. The `Retry-After` header describes when to try the write request again.
- `500` **Internal server error**: Default HTTP status for an error.

> - `503` **Service unavailable**: Server is temporarily unavailable to accept writes. The `Retry-After` header describes when to try the write again.

Reviewer: For each response codes, maybe provide some instruction on what to do to resolve. For example:

- link to review the correct format for line protocol.
- link to how/where to retrieve the API token
- where to view resources
- how to split large payloads, by time or other filter?
- anything more helpful to trace internal server errors?

The `message` property of the response body may contain additional details about the error.

## Troubleshoot failures

> If you notice data is missing in your bucket, check the following:

Reviewer: ```suggestion
If you notice data is missing in your bucket, do the following:
```

> - Does the `message` property in the response body contain details about the error?
> - Do all lines contain valid syntax, e.g. [line protocol](/influxdb/cloud/reference/syntax/line-protocol/) or [CSV](/influxdb/cloud/reference/syntax/annotated-csv/)?
> - Do the data types match the [bucket schema](/influxdb/cloud/organizations/buckets/bucket-schema/)?
>   For example, did you attempt to write `string` data to an `int` field?
> - Do the timestamps match the precision parameter?

Reviewer: ```suggestion
- Check the `message` property in the response body for details about the error
- Verify all lines contain valid syntax, e.g. [line protocol](/influxdb/cloud/reference/syntax/line-protocol/) or [CSV](/influxdb/cloud/reference/syntax/annotated-csv/)
- Verify the data types match the [bucket schema](/influxdb/cloud/organizations/buckets/bucket-schema/)
  For example, did you attempt to write `string` data to an `int` field?
- Verify the timestamps match the precision parameter
```

Reviewer: maybe link to more info about timestamp precision
/influxdb/cloud/write-data/#timestamp-precision
or /influxdb/cloud/reference/cli/influx/write/

### Troubleshoot rejected points

InfluxDB may reject points even if the HTTP request returns "Success".
If some of your data did not write to the bucket, check for [field type](/influxdb/cloud/reference/key-concepts/data-elements/#field-value) differences between the missing data point and other points that have the same [series](/influxdb/cloud/reference/key-concepts/data-elements/#series).

> For example, did you attempt to write `string` data to an `int` field?

Reviewer: duplicate line 73...prob can drop one.
Thoughts about rewording "Troubleshoot rejected points" to "Resolve rejected points" w concrete steps to resolve?

Author: That change is in the follow-up PR, for when `rejected_points` is live.

Author: Duplicated that because an explicit bucket schema will reject the batch if a point has a schema conflict, whereas an implicit schema will silently drop the point. I'll try to expand on that.
